TicTacToe.java

```diff
@@ -0,0 +1,52 @@
+import java.util.*;
+
+public class TicTacToe {
+
+  public static boolean checkCol(int x_coord, int[][] board) {
+    return (board[0][x_coord] == board[1][x_coord]) && (board[1][x_coord] == board[2][x_coord]);
+  }
+
+  public static boolean checkRow(int y_coord, int[][] board) {
+    return (board[y_coord][0] == board[y_coord][1]) && (board[y_coord][1] == board[y_coord][2]);
+  }
+
+  public static boolean checkRightDiagonal(int x_coord, int y_coord, int[][] board) {
+    if(board[0][0] == board[1][1] && board[1][1] == board[2][2]) return true;
+    else if(x_coord != y_coord) return false;
+  }
+
+  public static boolean checkLeftDiagonal(int x_coord, int y_coord, int[][] board) {
+    if(board[0][2] == board[1][1] && board[1][1] == board[2][0]) return true;
+    else if(x_coord != 2 - y_coord) return false;
+  }
+
+  public static void main(String[] args) {
+    int[][] board = new int[3][3];
+    int turn = 0;
+    int winner = 0;
+
+    while(turn < 9 && winner == 0) {
+      int player_turn = turn % 2 + 1;
+      Scanner sc = new Scanner(System.in);
+      System.out.print("Player " + player_turn + ": ");
+
+      String[] user_input = sc.nextLine().split(",");
+      int x_coord = Integer.parseInt(user_input[0]);
+      int y_coord = Integer.parseInt(user_input[1]);
+
+      if(board[y_coord][x_coord] == 0) {
```

Comment:
cleaner to do the following:
```
if(board[y_coord][x_coord] != 0) {
  System.out.println("That place is chosen by other player. Try other place.");
  continue;
}

board[y_coord][x_coord] = player_turn;
turn++;
```

Reply:
pattern to remember:
```
guard clause followed by core logic

guard clause - what cases do u want to exclude from going to core logic? remove it here
core logic
```

this ensures that unwanted cases are presented before the logic. with if/else statements, u can flip the position (which is not what we want) that's why we try to keep away from if/else. we try to keep the code as consistent as possible and so using guard clauses instead of if/else is preferrable.

For example:
if/else implementation:
```
if(board[y_coord][x_coord] == 0) { // case that we want
} else { // case that we don't want
}
```

guard clause implementation:
```
if(board[y_coord][x_coord] != 0) // skip logic here

// core logic here
```

in guard-clause implementation, developers are forced to put unwanted cases before the core logic since we want the code to never hit the core logic for unwanted cases